bin/reformat-files.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Find all JS and JSX files in the project that have the @format tag
+ * and reformat them with Prettier. Useful when upgrading Prettier to
+ * a newer version.
+ * @format
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const glob = require( 'glob' );
+const ignore = require( 'ignore' );
+const docblock = require( 'jest-docblock' );
+const prettier = require( 'prettier' );
+
+// Load ignore file
+const ignoreFile = fs.readFileSync( '.eslintignore', 'utf-8' );
+const ig = ignore().add( ignoreFile );
+
+// List .js and .jsx files in the current directory
+const files = glob.sync( '**/*.{js,jsx}', {
+	ignore: 'node_modules/**', // ignore node_modules by default
+	nodir: true,
+} );
+
+files.forEach( file => {
+	// skip ignored files
+	if ( ig.ignores( file ) ) {
+		return;
+	}
+
+	const text = fs.readFileSync( file, 'utf8' );
+
+	// skip if the first docblock doesn't have @format tag
+	const shouldFormat = 'format' in docblock.parse( text );
+	if ( ! shouldFormat ) {
+		return;
+	}
+
+	const formattedText = prettier.format( text, {} );
+
+	// did the re-formatting change anything?
+	if ( formattedText === text ) {
+		return;
+	}
+
+	fs.writeFileSync( file, formattedText );
+	console.log( `Prettier reformatted file: ${ file }` );
+} );
```

Comment:
Do you think it would be worth splitting up into several processes here? I wouldn't normally ask except that it seems like it would be quite easy with `child_process.spawn()`

Reply:
Personally i'd rather have this process be 5x slower than introduce parallelization and the complexities that brings with it.

Why?  Because its the sort of script we will run a handful of times per year and I don't mind waiting a few minutes.  Thats just my 2c though